### ISSUE TITLE
Thumbnails not available for files in symbolic links to network paths

### ISSUE THREAD

**Issue by DorCoMaNdO** [Xanashi/Icaros#22]
As I just posted over on the forums: https://shark007.net/forum/Thread-New-Release-3-2-1-Final?pid=430#pid430

Thumbnails do not get properly generated for files in symbolic links to network paths.

I recently built a NAS and moved a bunch of media to it, Windows Explorer properly generates thumbnails (both via and without Icarus) for media viewed through the network path (\\address\share\...) or through a mapped network drive, but NOT through a symbolic link to a network path (mklink /D "D:/Media" "\\address\), I primarily use the symbolic link to access my media, I saw that the old change logs (version 2.0.0) mentioned a fix for thumbnails on network drives, could this be addressed for symbolic links to network paths as well?
Windows Explorer does manage to generate thumbnails to certain JPG files, but not all, and all other media has no thumbnails (but the default ones).

I use Windows 8.1 but this is an issue that's been reported online as far back as Windows 7 and as far forward as 10.

If this does get fixed, but image formats remain broken (since they're not handled by Icaros), would it be possible for Icaros to generate thumbnails the same way Windows does for images? (mostly regarding gifs, use the same frame Windows picks for the thumbnail).

**Comment by clsid2** [comment]
I suspect Windows might be blocking this for security reasons. Microsoft has tightened security in Windows 8 and even more in 10.

The fact that the standard thumbnail plugin also doesn't work could mean that Explorer might not even be loading Icaros for such symlinked files. So solving it would then be impossible.

**Comment by DorCoMaNdO** [comment]
> The fact that the standard thumbnail plugin also doesn't work could mean that Explorer might not even be loading Icaros for such symlinked files.

But as I mentioned, it does, but it's highly limited, seems to only work for certain JPG files, likely based on compression/quality/bit depth, I have a bunch of family photos from a really old camera, those all seem to work, some downloaded digital drawings also work, but the vast majority of media does not.

I'll also specify that it's the directories being symlinked, not individual files, but symlinks to individual files (which I tried in the past) also does not generate thumbnails, even for files on local drives, however symlinks and junctions (both /D and /J) on local drives both generate thumbnails for files inside. Hardlinks to individual files work fine.

I'll also reiterate that Icaros does generate thumbnails over the network, as long as it's at a direct network path or mounted network drive, but not sym links, so this doesn't seem like a security change, this seems more like a symlink issue than a network issue, which can hopefully be addressed.

**Comment by Xanashi** [comment]
Hi DorCoMaNdO,

I tried my best to reproduce this issue without having access to a local network path, and so far haven't been able to recreate the bug.
I tried to setup a VM in VirtualBox, and create a symlink to the Shared Folder which simulates a network path, but thumbnailing still worked fine, so I assume it's not enough. 
(I used the following command to create the link: 
mklink /D "C:\Users\VMUser\Desktop\Symlink" "\\VBOXSVR\Shared_files\MKV Samples")

If I can somehow manage to reproduce the issue, I can probably tell you more, but there is a good chance that the issue is Windows itself as clsid2 mentioned above. 
The problem is MS decides when and where shell extensions are called by Explorer. The shell extensions themselves are just passive modules waiting to be called and used by Windows. So if MS decided that some network paths should not call third party shell extensions (most likely for security reasons), or maybe only use cached thumbnails, then there is not much I can do. 

Reproducing the bug would be useful in knowing for sure though.

**Comment by clsid2** [comment]
Use [ProcessMonitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) and make it filter for **Path contains = icaros**. Then you will know if it gets loaded after selecting the symlink in Explorer.

**Comment by DorCoMaNdO** [comment]
Icaros appears to always be loaded by explorer, in local folders, remote drives, remote paths, and remote symlinks.
In the thread reporting a similar issue to this since Windows 7, there were some mixed reports of newer Windows 10 versions fixing whatever issue this is, I'm currently on 8.1 and due to my current setup I cannot upgrade to Windows 10 as a solution for this issue.
Would you mind re-running your VM test on Windows 8.1? You can get an ISO from "TechBench by WZT".

**Comment by vvirtues** [comment]
not sure if this is related, but OneDrive files seem to do the same thing, with the exception of one `.jpg` (see image). putting the files in debug works for thumbnailing via icaros (https://cdn.pinglik.eu/u/files/vZ4shN0yZ9pqVcyoMQunc0rGaNOKW8WNhGjQ), but when using thumbnailing via explorer (https://cdn.pinglik.eu/u/files/wcl0v8FQpZKzWAHrfOgQU78QABzqxYugcccw), the provider is icaros but it doesnt seem to inject/load
https://cdn.pinglik.eu/u/files/HJWvHeAReAQQQyxTbR0mZ40kw1icpmDqt9my

**Comment by DorCoMaNdO** [comment]
Yeah it's exactly the same for me, and yes as I also mentioned some jpg files do work but the vast majority don't.
However in my case for some reason if I use direct network paths it works fine, if I use mapped paths it does not.
